_posts/2018-01-04-notification-channels-android-o.md

```diff
@@ -0,0 +1,129 @@
+
+Now, it's pretty easy to target API 26, but there's no gurantee that your app will function as it used to by just changing the `targetSdkVersion` value in your `build.gradle` file.
+
+This post looks specifically at one of the new things introduced in Android Oreo, and how targetting API 26 will affect your app, and notifications - notification channels (or categories).
+
+TL;DR
+-----
+
+Your app will stop sending notifications to the users if you target API 26 and you don't implement notification channels.
+
+What are Notification Channels?
+-----
+
+In Android O, notification channels are somewhat like groups or categories of notifications - say you're building a social networking app, the channels can be "activity" - like likes, or comments on your posts, "messages", etc. In fact, in the settings of your app, the user sees channels as "categories" as seen in the screenshot below.
+
+<!-- Insert notification settings screenshot here -->
+<p align="center">
+	<img src="/img/notification-channels-android-o.png"
+  alt="Android Studio showing app:fontFamily as error">
+</p>
+
+
+
+With the introduction of notification channels, users can get a fine control of what they want to be notified about - they can specifically turn off notifications for a certain channel, specify the importance as well as the preferred sound for a particular category of notifications, whether or not to override DND (Do not disturb)
+
+Implementing Notification Channels
+-----
+
+### Creating notification channels
+
+These notification channels are created at runtime, so they can be dynamically changed to fit the requirements of your the user. For example, if a user subscribes to group chat for example, you can create a channel matching that particular group, and when they leave the group, you can remove the notification channel.
+
+So how then does one add channels to notifications? At the simplest level, one needs to create a channel with a unique name and ID and a level of importance. The uniqueness of the name and ID only refers to your app package. The notification manager then does the creation of the channel with whatever settings you specify. Typically the creation code looks like:
+
+```kotlin
+val privateMessagesChannel = NotificationChannel(
+                    PRIVATE_MESSAGES_CHANNEL_ID,
+                    context.getString(R.string.pm_channel_name),
+                    NotificationManager.IMPORTANCE_DEFAULT)
+
+notificationManager.createNotificationChannel(privateMessagesChannel)
+```
+It's worthy of note that you can also create channel groups, and this is useful in scenarious where you have multiple accounts or profiles, so you can group the notification channels per account/profile.
```

Comment:
_scenarious_ 👉🏾 _scenarios_

Reply:
_It's worthy of note that you can also create channel groups, and this is useful in scenarious where you have multiple accounts or profiles, so you can group the notification channels per account/profile._

_It's worthy of note that you can also create channel groups. This is useful in scenarios where you have multiple accounts or profiles so that the notification channels can be grouped per account/profile._ 🤔

Reply:
🙈 typos everywhere. Thanks 🔒